Danalock V3

smulle48 opened the pull request:
Danalock support added

Conflicting files fixed :-)

Instead of reading the state in deCONZ GUI, I just added it to its own group in Phoncon. Removed one of the batteries and waited a few seconds. Then installed the battery again and turned the group in Phoscon on and off a few times

Michaelnorge commented:
Hello.
Just a maybe stupid question for stupid people like me ;-) :
I am using Raspberry Pi headless with Conbee/Phoscon and ioBroker.
How i can get Danalock supported in Phoscon?
I really do not understand it well.

Should I install this version in ioBroker?: https://github.com/smulle48/deconz-rest-plugin

smulle48 commented:
> Hello.
> Just a maybe stupid question for stupid people like me ;-) :
> I am using Raspberry Pi headless with Conbee/Phoscon and ioBroker.
> How i can get Danalock supported in Phoscon?
> I really do not understand it well.
> 
> Should I install this version in ioBroker?: https://github.com/smulle48/deconz-rest-plugin

The code is already merged into the master branch, so if you clone the master git, and build it yourself, you got danalock support as a light bulb

Smanar commented:
This version will be available soon, you can just wait for some days
And there is version already compiled > http://deconz.dresden-elektronik.de/raspbian/beta/

Michaelnorge commented:
@Smanar So do I just need to have v79? 
I installed it already yesterday but I am still not able to get the danalock to work :-(
I´m getting my versions fm here: http://deconz.dresden-elektronik.de/raspbian/beta/?C=M;O=D

Smanar commented:
Ha ?
Perhaps was too late, but according to date, it s possible having it in 79, I don't find information about it, but you will have more information in some days.
If you realy are hurried, you can too use the smulle48  method to compile it yourself.

Michaelnorge commented:
Thank you for informations, I think i will wait :-)

Michaelnorge commented:
Ok, one last question, because I´m confused: I need the GUI of Deconz, right? Because I´m using headless.
I know I´m annoying ;-))

Smanar commented:
Lol np, from that I m read, yes it s easier with the GUI but look at the first comment

> Instead of reading the state in deCONZ GUI, I just added it to its own group in Phoncon. Removed one of the batteries and waited a few seconds. Then installed the battery again and turned the group in Phoscon on and off a few times.

Michaelnorge commented:
Yeah weekend - now I have time!
Look @Smanar , I don´t know anything about Linux. The only thing I can do is to copy and paste commands.
The last time I had something with "programming" to do was on Basic on my Commodore C64 :-))))).

In line nr. 3 from @smulle48 instruction I´m getting this error:
`pi@raspberrypi:~ $ sudo git clone https://github.com/dresden-elektronik/deconz-rest-plugin.git
fatal: Zielpfad 'deconz-rest-plugin' existiert bereits und ist kein leeres Verzeichnis.`

What should I do?

smulle48 commented:
> Yeah weekend - now I have time!
> 
> Look @Smanar , I don´t know anything about Linux. The only thing I can do is to copy and paste commands.
> 
> The last time I had something with "programming" to do was on Basic on my Commodore C64 :-))))).
> 
> 
> 
> In line nr. 3 from @smulle48 instruction I´m getting this error:
> 
> `pi@raspberrypi:~ $ sudo git clone https://github.com/dresden-elektronik/deconz-rest-plugin.git
> 
> fatal: Zielpfad 'deconz-rest-plugin' existiert bereits und ist kein leeres Verzeichnis.`
> 
> 
> 
> What should I do?
> 
> 
> 
> And it´s one thing more: 
> 
> The description says: 
> 
> 
> 
> >  I just added it to its own group in Phoncon. Removed one of the batteries and waited a few seconds. Then installed the battery again and turned the group in Phoscon on and off a few times.
> 
> 
> 
> But what kind of group? In Phoscon I can create a group by my self, but will it come up itself?

It means that the directory you try to clone the files into, already exist.

Try deleting the folder first with the command:

sudo rm -rf deconz-rest-plugin 

When the lock is included in Phoscon, you can't control it before you create a group.

On the front page, create a group (the button on the top left) and call it something like lock or so. Then you add the "bulb".

After that, you can turn the group on and off, thereby controlling the lock :-)

Michaelnorge commented:
Ok, I understand now  @smulle48 .
I deleted the folder and it was working now. But again it stucks, at point 7:
`pi@raspberrypi:~/deconz-rest-plugin $ sudo mv libde_rest_plugin.so /usr/share/deCONZ/plugins/libde_rest_plugin.so
mv: der Aufruf von stat für 'libde_rest_plugin.so' ist nicht möglich: Datei oder Verzeichnis nicht gefunden`

smulle48 commented:
> Ok, I understand now  @smulle48 .
> 
> I deleted the folder and it was working now. But again it stucks, at point 7:
> 
> `pi@raspberrypi:~/deconz-rest-plugin $ sudo mv libde_rest_plugin.so /usr/share/deCONZ/plugins/libde_rest_plugin.so
> 
> mv: der Aufruf von stat für 'libde_rest_plugin.so' ist nicht möglich: Datei oder Verzeichnis nicht gefunden`
> 
> 

Ahh, yeah. The file is compiled outside the deconz-rest-plugin folder, so just do a "cd .." command first to move outside the folder :-) (remember the space between cd and ..)

Michaelnorge commented:
Yes that helped. Thank you so much, both of you!!!!

sembeeuk commented:
What is the trick to get the lock in to Phoscon? I have updated to .79, but when I press the button on the lock to get it to start flashing green and do add light in the app, nothing happens. Have I missed something? Can someone outline the complete steps please?

smulle48 commented:
> What is the trick to get the lock in to Phoscon? I have updated to .79, but when I press the button on the lock to get it to start flashing green and do add light in the app, nothing happens. Have I missed something? Can someone outline the complete steps please?

I don't think it's included in .79 so you have to build it yourself from source

sembeeuk commented:
> > What is the trick to get the lock in to Phoscon? I have updated to .79, but when I press the button on the lock to get it to start flashing green and do add light in the app, nothing happens. Have I missed something? Can someone outline the complete steps please?
> 
> I don't think it's included in .79 so you have to build it yourself from source

Definitely in .79. The issue has been closed: https://github.com/dresden-elektronik/deconz-rest-plugin/issues/520 and it is in the release notes. https://github.com/dresden-elektronik/deconz-rest-plugin/tree/V2_05_79

Michaelnorge commented:
Es war mir bis dato nicht möglich, das Danalock in ioBroker einzubinden. Ich konnte nach dankbarer Hilfe die v79 installieren. Allerdings findet Deconz nicht das Schloß. 

Mir kam die Idee, daß es evtl. daran liegt, daß ich das Schloss bereits in einer vorherigen Version von Deconz versucht habe, einzubinden?!

Das Phänomen hatte ich schon einmal, als ich IKEA-Schalter eingebunden hatte. Die wurden zwar nicht in der Phoscon-App aufgeführt, waren in ioBroker allerdings vorhanden und konnten auch benutzt werden.

Wenn also das Danalock bereits eingebunden ist und nicht in der Phoscon-App erscheint, dann ist es ja bei der v79 keine "neue" Hardware und wird nicht gefunden - weil eben schon vorhanden!

Kann man bei einer Headless irgendwie manuell Einträge auf dem Conbee-Stick löschen?

Grüße
Michael
PS: meine Englischkenntnisse reichen leider nicht, daher auf deutsch ;-)

Mimiix commented:
> Es war mir bis dato nicht möglich, das Danalock in ioBroker einzubinden. Ich konnte nach dankbarer Hilfe die v79 installieren. Allerdings findet Deconz nicht das Schloß. 
> 
> Mir kam die Idee, daß es evtl. daran liegt, daß ich das Schloss bereits in einer vorherigen Version von Deconz versucht habe, einzubinden?!
> 
> Das Phänomen hatte ich schon einmal, als ich IKEA-Schalter eingebunden hatte. Die wurden zwar nicht in der Phoscon-App aufgeführt, waren in ioBroker allerdings vorhanden und konnten auch benutzt werden.
> 
> Wenn also das Danalock bereits eingebunden ist und nicht in der Phoscon-App erscheint, dann ist es ja bei der v79 keine "neue" Hardware und wird nicht gefunden - weil eben schon vorhanden!
> 
> Kann man bei einer Headless irgendwie manuell Einträge auf dem Conbee-Stick löschen?
> 
> Grüße
> Michael
> PS: meine Englischkenntnisse reichen leider nicht, daher auf deutsch ;-)

Hi Michael,

I let your comment stay, but any further comments should be placed in English or will be removed.

Google translate would be fine too :)

Thank you,

Dennis

smulle48 commented:
> Es war mir bis dato nicht möglich, das Danalock in ioBroker einzubinden. Ich konnte nach dankbarer Hilfe die v79 installieren. Allerdings findet Deconz nicht das Schloß. 
> 
> 
> 
> Mir kam die Idee, daß es evtl. daran liegt, daß ich das Schloss bereits in einer vorherigen Version von Deconz versucht habe, einzubinden?!
> 
> 
> 
> Das Phänomen hatte ich schon einmal, als ich IKEA-Schalter eingebunden hatte. Die wurden zwar nicht in der Phoscon-App aufgeführt, waren in ioBroker allerdings vorhanden und konnten auch benutzt werden.
> 
> 
> 
> Wenn also das Danalock bereits eingebunden ist und nicht in der Phoscon-App erscheint, dann ist es ja bei der v79 keine "neue" Hardware und wird nicht gefunden - weil eben schon vorhanden!
> 
> 
> 
> Kann man bei einer Headless irgendwie manuell Einträge auf dem Conbee-Stick löschen?
> 
> 
> 
> Grüße
> 
> Michael
> 
> PS: meine Englischkenntnisse reichen leider nicht, daher auf deutsch ;-)

Hi Michael

Try to reset your Danalock by pressing the reset button on the lock 10 times. Then wait until in restart.

Then, in Phoscon, search for new light and then use the: Gerät -> Smart Home -> Z-Wave -> Verbinden inside the Danalock App.

I have no idea why they haven't changed the Z-Wave text to ZigBee :-)

Then your lock should show up as a new light in Phoscon.

Michaelnorge commented:
Hello again!
Nope, no chance to get it to work. I tried everything.
Are there maybe differences between the different versions of Danalock?
I am using Danalock v3 SCANDI.

My lock using firmware 0.16.0

smulle48 commented:
> Hello again!
> 
> Nope, no chance to get it to work. I tried everything.
> 
> Are there maybe differences between the different versions of Danalock?
> 
> I am using Danalock v3 SCANDI.
> 
> 
> 
> My lock using firmware 0.16.0
> 
> 
> 
> 

If it's the zigbee version you got, there are no differences in the lock itself. The only difference is how it's mounted to the lock itself.

How far from the conbee do you have the lock? You could try to move it next to your conbee to see if it makes any difference

ebaauw commented:
I fear this PR has some unintended side effects on the Xiaomi vibration sensor, see #3151.

niklerus commented:
> chance to get it to work. I tried everything.
> Are there maybe differences between the different versions of Danalock?
> I am using Danalock v3 SCANDI.
> 
> My lock using firmware 0.16.0

@Michaelnorge , did you ever get this to work?

smulle48 commented:
> > chance to get it to work. I tried everything.
> 
> > Are there maybe differences between the different versions of Danalock?
> 
> > I am using Danalock v3 SCANDI.
> 
> > 
> 
> > My lock using firmware 0.16.0
> 
> 
> 
> @Michaelnorge , did you ever get this to work?

Does your lock flash when you try to include it?

niklerus commented:
Asking me or Michaelnorge?
As I recall the green light was flashing when pairing.
I can add it to deconz but doesn't show up in Phoscon.
Missing name, vendor and model info, but node properties shows Home Automation Door Lock.
I can lock and unlock in Deconz, lock status follows.
Not entirely clear to me if I need to do your inital steps described in the beginning of this thread or if that can be omitted when on deconz V2_05_79 or later? I'm on  ..81

Michaelnorge commented:
@smulle48 It flashes green if I try to get it connect, but nothing else happens.
I just found out that when Phoscon is in "connect mode" the green diode is flashing for a shorter time than when Phoscon is not.
 
I tried it last time yesterday with the newest version of deconz (v82).

So: I never get it to work.

Michaelnorge commented:
Like @niklerus told the Danalock never shows up in the Phoscon app.
I can´t confirm if the lock is added in deconz because I´m using a headless version on a raspberry pi.

I've seen that @niklerus is coming from sweden, I am from norway.
So is it possible that the "Scandi-version" using another firmware??

Michaelnorge commented:
@smulle48 Right now I have seen your last comment. I will take down the lock and hold it right beside the deconz dongle.
Don't think it will help, but I give it a try. The distance right now is about 3 meters.

niklerus commented:
@Michaelnorge , yes we probably have the same hardware.
I suspect you might have the lock paired as I have but you can't verify it in deconz.
I used phoscon interface to pair but never got a reply or confirmation. Was a bit suprised when I saw it was connected in deconz and that I could control it.
Have been using it with Home assistant Rp4 and Smartthings so far but I want to remove that, less hardware and Smartthings sometimes drops the Danalock, requiering a new setup process.

smulle48 commented:
> Like @niklerus told the Danalock never shows up in the Phoscon app.
> 
> I can´t confirm if the lock is added in deconz because I´m using a headless version on a raspberry pi.
> 
> 
> 
> I've seen that @niklerus is coming from sweden, I am from norway, So I guess he is using the Scandi-Version like me.
> 
> Is it possible that the "Scandi-version" using another firmware??
> 
> 
> 
> 

The only difference between the different versions is just how it's mounted on the lock. The Danalock itself is the same with same firmware and everything.

Have you tried to exclude/reset it first?

Michaelnorge commented:
@smulle48 I had to drive to the shop because the batteries were empty again - after two month using that lock!

Yes of course, I made resets a couple of time, I have an Danabridge as well and disconnected it, so it's not in use anymore.
Beside I'm running one of these IKEA-bridges, I turned it of in case it makes inteferences with the deconz.

But nothing helped. 

And now I have removed the lock and held it directly to the Deconz during the pairing process - Nothing!

smulle48 commented:
> @smulle48 I had to drive to the shop because the batteries were empty again - after two month using that lock!
> 
> Yes of course, I made resets a couple of time, I have an Danabridge as well and disconnected it, so it's not in use anymore.
> Beside I'm running one of these IKEA-bridges, I turned it off in case it makes inteferences with the deconz.
> 
> But nothing helped.
> 
> And now I have removed the lock and held it directly to the Deconz during the pairing process - Nothing!

If you got access the the box by ssh, you can try the following command. Then you should get a lot of debug info :-)

Then you can try to include the lock and see what happens.

sudo deCONZ --auto-connect=1 --dbg-error=1 --http-port=8484 --ws-port=20877 --upnp=0 -platform minimal --dbg-aps=2 --dbg-info=2 --dbg-zcl=2 --dbg-zdp=2 --dbg-http=2

Michaelnorge commented:
Hei again and good morning, @smulle48 
I started that command and there is a lot of infos like you said.

But will it stop itself sometime?

I'm getting a lot of "reconnect network failed, try later" messages.

`06:06:16:085 dev /dev/ttyS0 (/dev/ttyS0)
06:06:17:043 wait reconnect 14 seconds
06:06:18:043 wait reconnect 13 seconds
06:06:18:543 failed to reconnect to network try=4
06:06:19:043 wait reconnect 12 seconds
06:06:19:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:06:20:043 wait reconnect 11 seconds
06:06:21:043 wait reconnect 10 seconds
06:06:22:042 wait reconnect 9 seconds
06:06:23:043 wait reconnect 8 seconds
06:06:23:543 failed to reconnect to network try=5
06:06:24:043 wait reconnect 7 seconds
06:06:25:043 wait reconnect 6 seconds
06:06:26:042 wait reconnect 5 seconds
06:06:27:043 wait reconnect 4 seconds
06:06:28:042 wait reconnect 3 seconds
06:06:28:542 failed to reconnect to network try=6
06:06:29:042 wait reconnect 2 seconds
06:06:29:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:06:30:043 wait reconnect 1 seconds
06:06:30:141 COM: /dev/ttyACM0 : ConBee II (0x1CF1/0x0030)
06:06:30:141 COM: /dev/ttyAMA0 :  (0x0000/0x0000)
06:06:30:141 COM: /dev/ttyS0 :  (0x0000/0x0000)
06:06:30:141 dev /dev/ttyS0 (/dev/ttyS0)
06:06:30:141 auto connect com /dev/ttyS0
06:06:30:146 COM check bootloader
06:06:32:153 COM timout query bootloader, assume application
06:06:32:153 Serial com connected
06:06:33:542 try to reconnect to network try=7
06:06:33:579 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:33:579 device state timeout ignored in state 2
06:06:36:542 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:36:543 device state timeout ignored in state 2
06:06:38:542 try to reconnect to network try=8
06:06:39:543 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:06:39:543 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:39:543 device state timeout ignored in state 2
06:06:42:542 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:42:542 device state timeout ignored in state 2
06:06:43:542 try to reconnect to network try=9
06:06:45:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:06:48:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:48:043 device state timeout ignored in state 2
06:06:48:543 try to reconnect to network try=10
06:06:49:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:06:50:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:06:53:043 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:53:043 device state timeout ignored in state 2
06:06:53:542 reconnect network failed, try later
06:06:54:043 networkState: CC_ReconnectNetwork
06:06:54:043 start reconnect to network
06:06:56:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:56:043 device state timeout ignored in state 2
06:06:59:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:59:042 device state timeout ignored in state 2
06:06:59:293 try to reconnect to network try=1
06:06:59:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:00:868 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:07:04:019 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:04:020 device state timeout ignored in state 2
06:07:04:542 try to reconnect to network try=2
06:07:04:612 COM: /dev/ttyACM0 : ConBee II (0x1CF1/0x0030)
06:07:04:612 COM: /dev/ttyAMA0 :  (0x0000/0x0000)
06:07:04:612 COM: /dev/ttyS0 :  (0x0000/0x0000)
06:07:04:612 dev /dev/ttyS0 (/dev/ttyS0)
06:07:06:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:07:06:042 MASTER kill cmd 0x08 (TIMEOUT)
06:07:09:043 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:09:043 device state timeout ignored in state 2
06:07:09:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:09:542 try to reconnect to network try=3
06:07:11:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 62
06:07:14:043 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:14:043 device state timeout ignored in state 2
06:07:14:542 try to reconnect to network try=4
06:07:16:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 62
06:07:19:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:19:042 device state timeout (handled)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:048 Serial com disconnected, reason: 4
06:07:19:048 device disconnected reason: 4, index: 1
06:07:19:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:19:542 failed to reconnect to network try=5
06:07:20:043 wait reconnect 15 seconds
06:07:20:542 void zmMaster::handleStateIdle(zmMaster::MasterEvent) not connected goto OFF state
06:07:20:542 device state timeout ignored in state 4
06:07:21:042 wait reconnect 14 seconds
06:07:22:042 wait reconnect 13 seconds
06:07:23:042 wait reconnect 12 seconds
06:07:24:042 wait reconnect 11 seconds
06:07:24:542 failed to reconnect to network try=6
06:07:25:043 wait reconnect 10 seconds
06:07:26:042 wait reconnect 9 seconds
06:07:27:043 wait reconnect 8 seconds
06:07:28:043 wait reconnect 7 seconds
06:06:53:542 reconnect network failed, try later
06:06:54:043 networkState: CC_ReconnectNetwork
06:06:54:043 start reconnect to network
06:06:56:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:56:043 device state timeout ignored in state 2
06:06:59:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:06:59:042 device state timeout ignored in state 2
06:06:59:293 try to reconnect to network try=1
06:06:59:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:00:868 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:07:04:019 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:04:020 device state timeout ignored in state 2
06:07:04:542 try to reconnect to network try=2
06:07:04:612 COM: /dev/ttyACM0 : ConBee II (0x1CF1/0x0030)
06:07:04:612 COM: /dev/ttyAMA0 :  (0x0000/0x0000)
06:07:04:612 COM: /dev/ttyS0 :  (0x0000/0x0000)
06:07:04:612 dev /dev/ttyS0 (/dev/ttyS0)
06:07:06:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 59
06:07:06:042 MASTER kill cmd 0x08 (TIMEOUT)
06:07:09:043 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:09:043 device state timeout ignored in state 2
06:07:09:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:09:542 try to reconnect to network try=3
06:07:11:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 62
06:07:14:043 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:14:043 device state timeout ignored in state 2
06:07:14:542 try to reconnect to network try=4
06:07:16:042 command queue EVENT_TIMEOUT, cmd: CMD_CHANGE_NET_STATE, seq: 62
06:07:19:042 void zmMaster::handleStateStatus(zmMaster::MasterEvent) EVENT_TIMEOUT
06:07:19:042 device state timeout (handled)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:043 MASTER kill cmd 0x08 (ERROR)
06:07:19:048 Serial com disconnected, reason: 4
06:07:19:048 device disconnected reason: 4, index: 1
06:07:19:542 Daylight now: goldenHour1, status: 160, daylight: 1, dark: 0
06:07:19:542 failed to reconnect to network try=5
06:07:20:043 wait reconnect 15 seconds
06:07:20:542 void zmMaster::handleStateIdle(zmMaster::MasterEvent) not connected goto OFF state
06:07:20:542 device state timeout ignored in state 4
06:07:21:042 wait reconnect 14 seconds
06:07:22:042 wait reconnect 13 seconds
06:07:23:042 wait reconnect 12 seconds
06:07:24:042 wait reconnect 11 seconds`

Like this it's rotating all the time.

Not really sure what happens :-)))

Shall I try to connect to danalock while the command is working or after?

smulle48 commented:
@Michaelnorge could be your headless deCONZ running in the background.

Try this command first, to turn that off so as they both are trying to access your conbee

sudo systemctl stop deconz

smulle48 commented:
@Michaelnorge You on discord?

Michaelnorge commented:
Sorry, I'm on what?


> @Michaelnorge You on discord?

Sorry, I'm on what?

Michaelnorge commented:
... It's running now...

smulle48 commented:
> Sorry, I'm on what?
> 
> 
> 
> 
> 
> > @Michaelnorge You on discord?
> 
> 
> 
> Sorry, I'm on what?
> 
> 

https://discord.gg/QFhTxqN

The official deCONZ chat 😀

Michaelnorge commented:
Now I am :-)

Smanar commented:
Your log are unreadables.
But it seem you have USB disconnection loop, so look at the others issues, you probaby need to update your firmware.

niklerus commented:
@smulle48 @Michaelnorge Hey guys! Please don't leave me out!
I'm in discord, trying to find you guys and see how this goes, hoping to pick up something to get my stuff working as well.

smulle48 commented:
> @smulle48 @Michaelnorge Hey guys! Please don't leave me out!
> 
> I'm in discord, trying to find you guys and see how this goes, hoping to pick up something to get my stuff working as well.
> 
> 

I'll be back this afternoon. Then we can catch up :-)

niklerus commented:
@smulle48 OK, cool!
My latest status, removed lock from deconz (right click, delete node), factory reset of danalock, added it to Danalock phone app again.
From phoscon in Home assistant, add light, in mobile app triggered pairing, nothing shows up in phoscon, but now connected in deconz, can check status, lock, unlock and now it actually shows model and manufacturer (in the cluster info attributes, in the right pane Node list there's still only MAC and NWK, rest is empty)

smulle48 commented:
> @smulle48 OK, cool!
> 
> My latest status, removed lock from deconz (right click, delete node), factory reset of danalock, added it to Danalock phone app again.
> 
> From phoscon in Home assistant, add light, in mobile app triggered pairing, nothing shows up in phoscon, but now connected in deconz, can check status, lock, unlock and now it actually shows model and manufacturer (in the cluster info attributes, in the right pane Node list there's still only MAC and NWK, rest is empty)

Try to remove one battery from the lock and wait a minute or two and then insert it again. Then it should work 😃

niklerus commented:
Thanks, but no luck.
You mean it should pop up under lights in phoscon after that procedure?

smulle48 commented:
> Thanks, but no luck.
> 
> You mean it should pop up under lights in phoscon after that procedure?

Catch me on discord (DrDice) 😃

Michaelnorge commented:
Could you get it to work, @niklerus ?

niklerus commented:
No, never shows up in Phoscon, everything looks fine i Denonz as far as I can tell.
Will see if I find smulle48 on discord now.

smulle48 commented:
> No, never shows up in Phoscon, everything looks fine i Denonz as far as I can tell.
> 
> Will see if I find smulle48 on discord now.

On Discord im called DrDice 😃

Mimiix commented:
@smulle48 Are you alive :D?

sembeeuk commented:
I have just updated to .83 and the latest version of the firmware for my Raspbee hat as released a few days ago. Then I deleted the lock from within the Deconz GUI and reset the lock itself. Went through the add light process and it appeared immediately. I think it was the removing the device from Deconz which was the key, as I had tried it again after doing the updates without success. 

The only annoying thing now is that it appears in Home Assistant as a light, rather than a lock, making integrations difficult. Anyone have thoughts on that?

sembeeuk commented:
Answering my own question, I can create a template to do it. Sorry about the dodgy formatting! 

 `lock:
platform: template
    name: Front Door
    value_template: "{{ is_state('light.front_door', 'on') }}"
    lock:
      service: light.turn_off
      data:
        entity_id: light.front_door
    unlock:
      service: light.turn_on
      data:
        entity_id: light.front_door   `

niklerus commented:
I also got it working after a number of attempts, but I removed it from within deconz every time.
What I did differently when I got it working was that I added the lock from deconz interface (permit join) instead of Phoscon.

My thoughts on integrations, use Node Red, I always do, works perfectly, easy to use, you get a good overview of all your automations and flows. And more or less no cumbersome scripting.

Smanar commented:
If you include the device throught deconz, it will be not visible in the api. But you can correct this with putting phoscon in permit join "add new light", and read again basic attribute (cluster 0x0000), with making the device awake if needed.

niklerus commented:
> If you include the device throught deconz, it will be not visible in the api. But you can correct this with putting phoscon in permit join "add new light", and read again basic attribute (cluster 0x0000), with making the device awake if needed.

It is present in Phoscon and I can control it as I wish in HA.
To be honest I did also start a "permit join" through the REST API and I'm not completely sure what did the trick, that or adding through deconz. 
But I'm certain I never did any "add new light" in phoscon

Smanar commented:
>I did also start a "permit join" through the REST API

Perfect ^^, it s like using "add new light" in phoscon.

patrick0392 commented:
> 
> 
> > Hello.
> > Just a maybe stupid question for stupid people like me ;-) :
> > I am using Raspberry Pi headless with Conbee/Phoscon and ioBroker.
> > How i can get Danalock supported in Phoscon?
> > I really do not understand it well.
> > Should I install this version in ioBroker?: https://github.com/smulle48/deconz-rest-plugin
> 
> The code is already merged into the master branch, so if you clone the master git, and build it yourself, you got danalock support as a light bulb.
> 
>     1. sudo apt update
> 
>     2. sudo apt install deconz-dev
> 
>     3. git clone https://github.com/dresden-elektronik/deconz-rest-plugin.git
> 
>     4. cd deconz-rest-plugin
> 
>     5. qmake && make -j2
> 
>     6. sudo mv /usr/share/deCONZ/plugins/libde_rest_plugin.so /usr/share/deCONZ/libde_rest_plugin.so.bak
>
>     7. sudo mv libde_rest_plugin.so /usr/share/deCONZ/plugins/libde_rest_plugin.so
> 
>     8. reopen deCONZ GUI

Hello,

i have also a problem at point 5
qmake && make -j2
-bash: qmake: Kommando nicht gefunden. ( qmake: this command not found)

so whats the problem i'm a noob @ linux ... need help...

smulle48 commented:
> > 
> 
> > 
> 
> > > Hello.
> 
> > > Just a maybe stupid question for stupid people like me ;-) :
> 
> > > I am using Raspberry Pi headless with Conbee/Phoscon and ioBroker.
> 
> > > How i can get Danalock supported in Phoscon?
> 
> > > I really do not understand it well.
> 
> > > Should I install this version in ioBroker?: https://github.com/smulle48/deconz-rest-plugin
> 
> > 
> 
> > The code is already merged into the master branch, so if you clone the master git, and build it yourself, you got danalock support as a light bulb.
> 
> > 
> 
> >     1. sudo apt update
> 
> > 
> 
> >     2. sudo apt install deconz-dev
> 
> > 
> 
> >     3. git clone https://github.com/dresden-elektronik/deconz-rest-plugin.git
> 
> > 
> 
> >     4. cd deconz-rest-plugin
> 
> > 
> 
> >     5. qmake && make -j2
> 
> > 
> 
> >     6. sudo mv /usr/share/deCONZ/plugins/libde_rest_plugin.so /usr/share/deCONZ/libde_rest_plugin.so.bak
> 
> >
> 
> >     7. sudo mv libde_rest_plugin.so /usr/share/deCONZ/plugins/libde_rest_plugin.so
> 
> > 
> 
> >     8. reopen deCONZ GUI
> 
> 
> 
> Hello,
> 
> 
> 
> i have also a problem at point 5
> 
> qmake && make -j2
> 
> -bash: qmake: Kommando nicht gefunden. ( qmake: this command not found)
> 
> 
> 
> so whats the problem i'm a noob @ linux ... need help...
> 
> 
> 
> 
> 
> 
> 
> 

The code is already merged into deCONZ so if you just update deCONZ to the latest version, you got it without compiling yourself :-)

patrick0392 commented:
So i have only to do 
Sudo apt update deconz?

Michaelnorge commented:
@patrick0392 Look here: [https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Update-deCONZ-manually](url)

patrick0392 commented:
okay thanks doest it work on your pi ?

patrick0392 commented:
So i have updated now... but i can not finde my lock ...

patrick0392 commented:
@sudo apt install deconz-dev

Fehl:1 http://phoscon.de/apt/deconz buster/main armhf deconz-dev armhf 2.05.81-raspbian-buster-stable
  404  Not Found [IP: 144.76.96.194 80]
E: Fehlschlag beim Holen von http://phoscon.de/apt/deconz/pool/main/d/deconz-dev/deconz-dev_2.05.81-raspbian-buster-stable_armhf.deb 404  Not Found [IP: 144.76.96.194 80]
E: Einige Archive konnten nicht heruntergeladen werden; vielleicht »apt-get update« ausführen oder mit »--fix-missing« probieren?

Smanar commented:
You don't need to update the dev package, this is for compiling yourself the code.
This device is already in deconz now, just need to update your deconz version.

Michaelnorge commented:
If you updated to the last version (at this moment deconz-2.05.85-qt5.deb | 16-Oct-2020 00:22) you should be able to connect Danalock without the workaround from smule48 :-)

Please let me know if you got it working!

smulle48 commented:
> If you updated to the last version (at this moment deconz-2.05.85-qt5.deb | 16-Oct-2020 00:22) you should be able to connect Danalock without the workaround from smulle48 :-)
> 
> 
> 
> Please let me know if you got it working!
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

Yeah

My code is already in the official code

patrick0392 commented:
Okay i have the new Version but it Dienst work ... :/

Smanar commented:
Can you see something in deconz ?

patrick0392 commented:
hello,

i have a big problem,.. i cant find anything... I have change my raspberry (pi3 to 4) and now i have this error... what can i do ??... 

`
pi@raspberrypi:~ $ sudo service deconz-gui status
● deconz-gui.service - deCONZ: ZigBee gateway -- GUI/REST API
   Loaded: loaded (/lib/systemd/system/deconz-gui.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Mon 2020-10-19 20:33:42 BST; 5s
  Process: 12507 ExecStart=/usr/bin/deCONZ --http-port=80 (code=exited, status=1/FAILURE)
 Main PID: 12507 (code=exited, status=1/FAILURE)
`pi@raspberrypi:~ $ sudo service deconz status
● deconz.service - deCONZ: ZigBee gateway -- REST API
   Loaded: loaded (/lib/systemd/system/deconz.service; enabled; vendor preset: e
   Active: active (running) since Mon 2020-10-19 20:37:47 BST; 5min ago
 Main PID: 344 (deCONZ)
    Tasks: 5 (limit: 4915)
   CGroup: /system.slice/deconz.service
           └─344 /usr/bin/deCONZ -platform minimal --http-port=80

Oct 19 20:37:47 raspberrypi systemd[1]: Started deCONZ: ZigBee gateway -- REST A
Oct 19 20:37:48 raspberrypi deCONZ[344]: QStandardPaths: XDG_RUNTIME_DIR not set
Oct 19 20:37:48 raspberrypi deCONZ[344]: libpng warning: iCCP: known incorrect s
Oct 19 20:37:49 raspberrypi deCONZ[344]: This plugin does not support propagateS
Oct 19 20:37:49 raspberrypi deCONZ[344]: This plugin does not support propagateS
Oct 19 20:37:50 raspberrypi deCONZ[344]: This plugin does not support propagate`

thanks for help
`

Smanar commented:
It s like you have both version running in same time, the headless and the GUI, just disable and stop one of them.

https://github.com/dresden-elektronik/deconz-rest-plugin#headless-support-for-linux

patrick0392 commented:
okay, i have done this 
$ sudo systemctl disable deconz-gui
$ sudo systemctl stop deconz-gui

now only deconz is active but i find nothing on the phoscon page...

SwoopX commented:
Can you please stop polluting this PR history with totally unrelated discussions?

@patrick0392 Please open a seperate issue for that or joind our Discord channel, thanks.

Mimiix commented:
As @SwoopX Said. 

If anyone wants this re-opend. Pm me on discord.
